governance/proposals/010-arbitrum-l1-l2-messaging.js

```diff
@@ -0,0 +1,215 @@
+  toL2 = GRANTS_CONTRACT_ADDRESS,
+  fromL2 = TIMELOCK_L2_ALIAS,
+}) => {
+  console.log(
+    'Proposal For Executing L1 to L2 Messaging Using Arbitrum Delayed Inbox (Retryable Tickets)'
+  )
+
+  const l2Network = await getL2Network(l2Provider)
+  const ethBridger = new EthBridger(l2Network)
+  const inboxAddress = ethBridger.l2Network.ethBridge.inbox
+
+  // token on L2
+  const L2TokenContract = new ethers.Contract(
+    tokenAddressL2,
+    ERC20_ABI,
+    l2Provider
+  )
+  const decimals = await L2TokenContract.decimals()
+
+  // check balance of sender on L2
+  const balanceOf = await L2TokenContract.balanceOf(fromL2)
+  const tokenAmount = ethers.utils.parseUnits('8200', decimals)
+
+  // Create an instance of the Interface from the ABIs
+  const erc20ContractInterface = new ethers.utils.Interface(ERC20_ABI)
+  const inboxContractInterface = new ethers.utils.Interface(INBOX_ABI)
+
+  // Encode the ERC20 Token transfer calldata
+  const transferCalldata = erc20ContractInterface.encodeFunctionData(
+    'transfer',
+    [toL2, tokenAmount]
+  )
+
+  /**
+   * Now we can query the required gas params using the estimateAll method in Arbitrum SDK
+   */
+  const l1ToL2MessageGasEstimate = new L1ToL2MessageGasEstimator(l2Provider)
+
+  const estimateAllParams = {
+    from: fromL1,
+    to: tokenAddressL2,
+    l2CallValue: 0,
+    excessFeeRefundAddress: fromL1,
+    callValueRefundAddress: fromL1,
+    data: transferCalldata,
+  }
+
+  /**
+   * The estimateAll method gives us the following values for sending an L1->L2 message
+   * (1) maxSubmissionCost: The maximum cost to be paid for submitting the transaction
+   * (2) gasLimit: The L2 gas limit
+   * (3) deposit: The total amount to deposit on L1 to cover L2 gas and L2 call value
+   */
+  const L1ToL2MessageGasParams = await l1ToL2MessageGasEstimate.estimateAll(
+    estimateAllParams,
+    await getBaseFee(l1Provider),
+    l1Provider
+  )
+  const gasPriceBid = await l2Provider.getGasPrice()
+  const ETHDeposit = L1ToL2MessageGasParams.deposit.toNumber() * 10 // I Multiply by 10 to add extra in case gas changes due to proposal delay
+
+  const params = [
+    estimateAllParams.to,
+    estimateAllParams.l2CallValue,
+    L1ToL2MessageGasParams.maxSubmissionCost.toString(), // maxSubmissionCost
+    estimateAllParams.excessFeeRefundAddress,
+    estimateAllParams.callValueRefundAddress,
+    L1ToL2MessageGasParams.gasLimit.toString(), // gasLimit
+    gasPriceBid.toString(), // maxFeePerGas
+    estimateAllParams.data,
+  ]
+
+  const inboxCalldata = inboxContractInterface.encodeFunctionData(
+    'createRetryableTicket',
+    params
+  )
+
+  const proposalName = `
+  # Transfer 8200 ARB To Fund Unlock Protocol’s Ecosystem via Grants Stack
+  
+  ### Goal of the proposal
+   This proposal requests to use 8200 ARB from the tokens given to Unlock Protocol DAO by ArbitrumDAO to fund the retroQF round on Grants Stack for projects building on Unlock Protocol.
```

Comment:
This is UP from what the snapshot proposal initially asked...

> Asking for 7k in ARB to help Fund Unlock Protocol’s Ecosystem via Grants Stack

Reply:
There is an explanation for why this is up from the initial 7k as well as a reference to the snapshot proposal included in the description